textile/chat-features.textile

```diff
@@ -322,20 +326,22 @@ Presence allows chat room users to indicate to others that they're online, as we
 ** @(CHA-PR3c)@ This specification point has been removed.
 *** @(CHA-PR3c1) This specification point has been removed.
 *** @(CHA-PR3c2)@ This specification point has been removed.
-** @(CHA-PR3d)@ @[Testable]@ If the room status is @Attaching@, then the @enter@ call will wait for the attach operation to succeed and then call the underlying presence @enter@ operation. It will throw an error if the attach fails.
+** @(CHA-PR3d)@ @[Testable]@ If the room status is @Attaching@, then the operation shall enter a @CHA-RL9@ pending room status operation. If this succeeds (i.e the room becomes @ATTACHED@), the library must invoke the underlying @presence.enter()@ call.
+** @(CHA-PR3h)@ @[Testable]@ If the room status is anything other than @Attached@ or @Attaching@, an @ErrorInfo@ using the @RoomInInvalidState@ error code from the "chat-specific error codes":#error-codes with a status code of @400@ must be thrown. The message shall explain that attach must be called first.
 ** @(CHA-PR3e)@ @[Testable]@ If the room status is @Attached@, then the @enter@ call will invoke the underlying @presence.enter()@ call.
-** @(CHA-PR3f)@ @[Testable]@ If the room status is @Detached@, an @ErrorInfo@ with code 40000, explaining that attach must be called first, will be thrown.
-** @(CHA-PR3g)@ @[Testable]@ For any other room status, an @ErrorInfo@ with code 40000 will be thrown.
+** @(CHA-PR3f)@ This specification point has been removed. It was superseded by @CHA-PR3h@.
+** @(CHA-PR3g)@ This specification point has been removed. It was superseded by @CHA-PR3h@.
 * @(CHA-PR10)@ Users may update their presence data. The behaviour depends on the current room status, as presence operations in a Realtime Client cause implicit attaches.
 ** @(CHA-PR10a)@ @[Testable]@ Users may choose to update their presence data, optionally providing custom data to update with. The overall presence data must retain the format specified in @CHA-PR2@.
 ** @(CHA-PR10b)@ This clause has been replaced by @CHA-PR10c - CHA-PR10g@.
 ** @(CHA-PR10c)@ This specification point has been removed.
 *** @(CHA-PR10c1)@ This specification point has been removed.
 *** @(CHA-PR10c2)@ This specification point has been removed.
-** @(CHA-PR10d)@ @[Testable]@ If the room status is @Attaching@, then the @update@ call will wait for the attach operation to succeed and then call the underlying presence @update@ operation. It will throw an error if the attach fails.
+** @(CHA-PR10d)@ @[Testable]@ If the room status is @Attaching@, then the operation shall enter a @CHA-RL9@ pending room status operation. If this succeeds (i.e the room becomes @ATTACHED@), the library must invoke the underlying @presence.update()@ call.
```

Comment:
```suggestion
** @(CHA-PR10d)@ @[Testable]@ If the room status is @Attaching@, then the operation should wait as per @CHA-RL9a@, once RoomStatus is @Attached@ as per @CHA-RL9b@ then the library must invoke the underlying @presence.update()@ call, otherwise throw error as per @CHA-RL9c@
```

Reply:
Will mark this as resolved and continue discussion on the one I've left open